packages/openapi3/src/openapi.ts

```diff
@@ -681,7 +681,8 @@ function createOAPIEmitter(program: Program, options: ResolvedOpenAPI3EmitterOpt
     currentEndpoint.description = shared.description;
     currentEndpoint.parameters = [];
     currentEndpoint.responses = {};
-    const visibility = http.getRequestVisibility(verb);
+    // TODO: this is probably not a good assumption?
+    const visibility = http.getRequestVisibility(program, shared.operations[0], verb);
```

Comment:
I think we need to calculate this separately, and potentially have different schemas, right?